plugins/kyverno/main.sh

```diff
@@ -1,4 +1,32 @@
 #! /bin/bash
+# Additional information on PolicyReport,ClusterPolicyReport: https://kyverno.io/docs/policy-reports/
+
+populate_policy_metadata() {
+  # to avoid overhead, collect only the last (most recent result) from the policyreport
+  report_json=$(kubectl get $KIND $report -n $namespace -o json | jq '.results |= [.[-1]]')
+  policy_name=$(echo $report_json | jq -r '.results[0].policy')
+
+  # use lookup table to minimize control plane load
+  if [[ ! "${policies[*]}" =~ "${policy_name}" ]]; then
```

Comment:
this might give false positives, per [here](https://stackoverflow.com/questions/3685970/check-if-a-bash-array-contains-a-value)

you could probably check sth like `if -z $policy_titles["$policy_name"]`

Reply:
This is a bit of a pain, "policy_title" and "policy_description" are allowed to be blank, so the only robust way to check is to keep a list of policies we've already retrieved from the k8s API and check that way.

Reply:
That being said, since all of the policy names are Kubernetes object names, they should never have whitespace in them. So I think this will be fine unless i'm missing something. I do want to clean this up a bit: i don't need to redeclare those variables and the function I introduced isn't as readable as just having some code duplication there IMO